osu.Game/Scoring/ScoreImporter.cs

```diff
@@ -99,7 +101,7 @@ public void PopulateMaximumStatistics(ScoreInfo score)
             if (score.MaximumStatistics.Select(kvp => kvp.Value).Sum() > 0)
                 return;
 
-            var beatmap = score.BeatmapInfo.Detach();
+            var beatmap = score.BeatmapInfo?.Detach();
```

Comment:
This one is dubious to me. In particular, `BackgroundBeatmapProcessor` can call this. It doesn't seem impossible that it can call it on a "dead" / "detached" score with no beatmap. If that happens, none of this will _fail_, but due to how `GetWorkingBeatmap()` works, dummy zero data will get inserted.

Background processing should likely not take place for those sorts of scores.

Reply:
That's a fair call, will change to throw.

Reply:
Changing to throw is fine and all, but my point was moreso that scores with null beatmap info could _feasibly be fed_ into `PopulateMaximumStatistics()` and we should possibly be doing something like the following:

```diff
diff --git a/osu.Game/BackgroundBeatmapProcessor.cs b/osu.Game/BackgroundBeatmapProcessor.cs
index b8c89d8822..41c31b6231 100644
--- a/osu.Game/BackgroundBeatmapProcessor.cs
+++ b/osu.Game/BackgroundBeatmapProcessor.cs
@@ -157,8 +157,12 @@ private void processScoresWithMissingStatistics()
             {
                 foreach (var score in r.All<ScoreInfo>())
                 {
-                    if (score.Statistics.Sum(kvp => kvp.Value) > 0 && score.MaximumStatistics.Sum(kvp => kvp.Value) == 0)
+                    if (score.BeatmapInfo != null
+                        && score.Statistics.Sum(kvp => kvp.Value) > 0
+                        && score.MaximumStatistics.Sum(kvp => kvp.Value) == 0)
+                    {
                         scoreIds.Add(score.ID);
+                    }
                 }
             });
 

```

Not nullchecking the beatmap won't _crash_ anything (assertion in debug / nullref in release will be try-caught), but it'll save us ugly log errors if this situation somehow does happen at some point.

Reply:
Makes sense, have applied the change.